Add changelog of YDB 23.3.17 release. (#1153)
ydb-platform/ydb: ydb/docs/en/core/changelog-server.md

# {{ ydb-short-name }} Server changelog

## Version 23.3.17 {#23-3-17}

Release date: January 5, 2024.

**Bugfixes:**

> * Fix `InFlightTxCount` become negative during `split/merge`.

Reviewer: ```suggestion
* Fix `InFlightTxCount` metric becoming negative during datashard split and merge.
```

* Allow compatibility between non-stable versions of different applications.

> * Various stability improvements (`segfault`, handle balancer and boot queue events in `StateInit`, data race when reporting issue counters, memory leak in TRopeArena).

Reviewer: split items in round brackets into separate bullet points

> * CPU over utilization in case of numerous topic consumers.

Reviewer: ```suggestion
* CPU overutilization in case of numerous topic consumers.
```

> * Fix import multiple converted csv columns.

Reviewer: isn't it about CLI?

> * Fix mem table merge when writing over committed deltas in tablets.

Reviewer: ```suggestion
* Fix memtable merge when writing over committed deltas in tablets.
```

> * Fix mem table merge when writing over committed deltas in tablets.

Reviewer: it's not clear how severe this is — were merges before the fix leading to incorrect data stored and responded?

* Check errors during `read_table_scan` and export.
* Throttle datashard log messages.

> **Features:**

Reviewer: why there are so many features on a minor release?

* Enable ANSI settings for generic query.

> * DML logging.

Reviewer: is this about audit logging? and is it enabled by default?

* Request unit trailing header for ScanQuery/ScanYql scripts.

> * Grpc KV api.

Reviewer: what about it?

* Change default metering mode of topics to request units.

> * Pqv0 support on server side.

Reviewer: what it is?
